### ISSUE TITLE
False positives for SCSS operators in media-feature-no-missing-punctuation

### ISSUE THREAD

**Issue by igoradamenko** [stylelint/stylelint#1754]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug with `@media` queries in SCSS when value of `max-width` (or other property) is not in parentheses.

> Which rule, if any, is this issue related to?

`media-feature-no-missing-punctuation`.

> What CSS is needed to reproduce this issue?

```
$content-size: 500px;
$side: 200px;

@media (max-width: $content-size + $side * 2) {
    position: relative;
    right: auto;

    max-width: none;
    padding: 0;
}
```

> What stylelint configuration is needed to reproduce this issue?

``` json
{
  "rules": {
    "media-feature-no-missing-punctuation": true
  }
}
```

> Which version of stylelint are you using?

`7.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

gulp-stylelint.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

``` shell
4:9   ×  Unexpected missing punctuation  media-feature-no-missing-punctuation
```

**Comment by evilebottnawi** [comment]
@igoradamenko thanks for feed

**Comment by jeddy3** [comment]
@evilebottnawi Would refactoring this rule to make use of `postcss-media-query-parser` help resolve this?

**Comment by evilebottnawi** [comment]
@jeddy3 today i try :smile:

**Comment by slavanga** [comment]
Another case of false positives occurs when using Sass functions (e.g round) in a media query:

```scss
@media (min-resolution: round($ratio * 96dpi)), (min-resolution: $ratio * 1dppx) {
	@content;
}

**Comment by jeddy3** [comment]
Deprecated rule.
